src/features/joinForms/actions/submitEmbeddedJoinForm.ts

```diff
@@ -0,0 +1,40 @@
+'use server';
+
+import Iron from '@hapi/iron';
+import { headers } from 'next/headers';
+
+import BackendApiClient from 'core/api/client/BackendApiClient';
+import { EmbeddedJoinFormData, EmbeddedJoinFormStatus } from '../types';
+
+export default async function submitJoinForm(
+  prevState: EmbeddedJoinFormStatus,
+  inputFormData: FormData
+) {
+  const headersList = headers();
+  const headersEntries = headersList.entries();
+  const headersObject = Object.fromEntries(headersEntries);
+  const apiClient = new BackendApiClient(headersObject);
+
+  const encrypted = inputFormData.get('__joinFormData')?.toString() || '';
+  const joinFormInfo = (await Iron.unseal(
+    encrypted,
+    process.env.SESSION_PASSWORD || '',
+    Iron.defaults
+  )) as EmbeddedJoinFormData;
+
+  const outputFormData: Record<string, string> = {};
+
+  joinFormInfo.fields.forEach((field) => {
+    outputFormData[field.s] = inputFormData.get(field.s)?.toString() || '';
+  });
+
+  await apiClient.post(
+    `/api/orgs/${joinFormInfo.orgId}/join_forms/${joinFormInfo.formId}/submissions`,
+    {
+      form_data: outputFormData,
+      submit_token: joinFormInfo.token,
+    }
+  );
+
+  return 'submitted';
```

Comment:
Mostly out of curiosity, but I'm wondering why this function always returns the same string - is it so unlikely that anything would go wrong that it makes no sense to, for example save the return value from the `POST` and depending on if it was ok, return either `"submitted"` or something like `"error"`?

Reply:
Yeah, the return value will be the new value for the state on the frontend (per the workings of React's new `useFormState()` hook) and proper error handling would be a good upgrade for this. Right now the error will result in a 500 and a generic error on the frontend which I thought was "ok" for MVP but sure should be improved.